core/image.cpp

```diff
@@ -1198,7 +1198,9 @@ void Image::expand_x2_hq2x() {
 	if (current != FORMAT_RGBA8)
 		convert(current);
 
-	if (used_mipmaps) {
+	// FIXME: This is likely meant to use "used_mipmaps" as defined above, but if we do,
+	// we end up with a regression: GH-22747
+	if (mipmaps) {
```

Comment:
@reduz I did not dig much to understand why it misbehaves, but it looks like fixing this apparent bug caused #22747.
All other methods set `used_mipmaps = has_mipmaps()` at the beginning and regenerate mipmaps if it was set, while here we check `mipmaps` which is always false (due to `clear_mipmaps()` above).

So the question is why generating mipmaps in `expand_x2_hq2x` causes #22747, and how to fix it.

Reply:
To be clear, this commit undoes a previous "fix" in https://github.com/godotengine/godot/commit/dec20a987b40e7625b38e520cc223723da9bae2a#diff-1cf7b15b355bb206717d5061d1768debR1201 which causes the bug.